src/tools/compiletest/src/runtest.rs

```diff
@@ -3356,26 +3464,7 @@ impl<'test> TestCx<'test> {
                 println!("normalized {}:\n{}\n", kind, actual);
             } else {
                 println!("diff of {}:\n", kind);
-                let diff_results = make_diff(expected, actual, 3);
-                for result in diff_results {
-                    let mut line_number = result.line_number;
-                    for line in result.lines {
-                        match line {
-                            DiffLine::Expected(e) => {
-                                println!("-\t{}", e);
```

Comment:
Okay this is weird, but apparently `println!` in `compiletest` is *not* equivalent to writing to `std::io::stdout()`, what's happening now is that the diff is printed *while the tests run*, and then the output *after the tests run* looks like this:
```
diff of stderr:


The actual stderr differed from the expected stderr.
```

Reply:
Maybe some kind of buffering?

Reply:
Looking at the implementation it seems like there should be no difference unless something is calling set_print -- I think libtest does this, though, so maybe that's causing problems here.

Reply:
Ahh, `compiletest` is getting that behavior from `libtest`, that's what I was missing, thanks!
Yeah so worst case we can buffer into a `String` and `println!` that instead of using `std::io::stdout`.

Reply:
Turns out the changes here are unnecessary, there's only `write_diff` caller, and I've just tested a fix, will open a PR shortly.